moveos/moveos-stdlib/moveos-stdlib/sources/object_storage.move

```diff
@@ -33,13 +34,31 @@ module moveos_std::object_storage {
     public fun remove<T: key>(this: &mut ObjectStorage, object_id: ObjectID): Object<T>{
         raw_table::remove<ObjectID, Object<T>>(*&this.handle, object_id)
     }
-    
+
+    const EObjectOwnerIsEmpty: u64 = 101;
     #[private_generic(T)]
     /// Add object to object store
-    public fun add<T: key>(this: &mut ObjectStorage, obj: Object<T>) {
+    public(friend) fun add<T: key>(this: &mut ObjectStorage, obj: Object<T>) {
+        assert!(option::is_some(&object::owner(&obj)), EObjectOwnerIsEmpty);
         raw_table::add<ObjectID, Object<T>>(*&this.handle, object::id(&obj), obj);
     }
 
+    #[private_generic(T)]
+    public fun transfer<T: key>(this: &mut ObjectStorage, obj: Object<T>, recipient: address) {
+        object::set_address_owner(&mut obj, recipient);
+        add(this, obj);
+    }
+
+    #[private_generic(T)]
+    public fun freeze_object<T: key>(this: &mut ObjectStorage, obj: Object<T>) {
+        add(this, obj);
```

Comment:
Is the operation of setting the Owner flag missing here? And also in the below fun ```share_object```.